DOCKER/docker-entrypoint.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -e
 
-if [ ! -d "$TMHOME/config" ]; then
+if [ ! -d "$TMHOME/config" ] && [ "$@" != "init" ]; then
```

Comment:
I think it's not necessary to check the input command here and it's a bit misleading. (The command will accidentally work, but it won't really safeguard what it's supposed to.)

One problem with parsing the input is that bash is just plain bad at it. You can run `tendermint init --log_level info` which is completely valid and the check here would miss it. (And we didn't even get to `tendermint init --home /otherfolder yet`.)

The `tendermint init` command doesn't overwrite anything, if run on an initialized directory. You don't have to worry about it.

Checking the directory is kind of nice, because if there's no config but someone runs `tendermint node`, you have a config at the ready. (Which is what we had with the static config too, before.)

If you really don't want the benefits that the wrapper is giving (tweaked config), you can always just work around the wrapper by `docker run --entrypoint /usr/bin/tendermint tendermint/tendermint init`.

If I understand this correctly, the main point of this check is that command like `tendermint show_node_id` don't initialize a config. The reality with the tendermint binary, that it's going to do it regardless: the way the config file is read, it will write a default config if you don't have any config dir and you only ran `tendermint show_node_id`. With this wrapper, at least it creates a complete configuration which I consider better, than the "default config, but no other files" approach of the code.

Last but not least, if a user mounted a volume, with a valid config, the directory check is enough to not overwrite it accidentally. So that works out too.

```suggestion
if [ ! -d "$TMHOME/config" ]; then
```

Reply:
Thanks for the extensive write up. I was back and forth on it. my main concern was that if ran `init` it wouldn't have any effect (because the sed defaults will run first and init doesn't overwrite existing config) and hence modified it in the last commit: https://github.com/tendermint/tendermint/pull/5087/commits/cdf5fde730e4f3336ce4f988a9322deaf90af3c3

Will revert that.